src/XrdTls/XrdTlsSocket.cc

```diff
@@ -640,7 +649,8 @@ XrdTls::RC XrdTlsSocket::Read( char *buffer, size_t size, int &bytesRead )
     // have to explicitly call SSL_connect or SSL_do_handshake.
     //------------------------------------------------------------------------
 
- do{int rc = SSL_read( pImpl->ssl, buffer, size );
+ do{ERR_clear_error();
```

Comment:
Frankly, the server does not have this issue as it seems to do quite well with post cleanup. I am still rather suprised it doesn't work for the client but then again the threading model is different. So, let's start with...

if (pImpl->isClient) ERR_clear_error();